base_layer/core/src/mempool/mempool_storage.rs

```diff
@@ -225,6 +230,17 @@ impl MempoolStorage {
             .reorg_pool
             .remove_reorged_txs_and_discard_double_spends(removed_blocks, new_blocks);
         self.insert_txs(removed_txs);
+
+        if !new_blocks.is_empty() {
+            self.last_seen_height = new_blocks.last().map(|block| block.header.height).unwrap_or(0 as u64);
+        } else if !removed_blocks.is_empty() {
+            self.last_seen_height = removed_blocks
+                .first()
+                .map(|block| block.header.height)
+                .unwrap_or(1 as u64) -
+                1;
+        } else {
+        }
```

Comment:
I don't quite get the significance of the different default return values. If they are insignificant we could cleanup the block with something like:

```suggestion
        if let Some(height) = new_blocks
            .last()
            .or(removed_blocks.first())
            .map(|block| block.header.height)
        {
            self.last_seen_height = height;
        }
```